app/controllers/application_controller.rb

```diff
@@ -17,21 +17,21 @@ class ApplicationController < ActionController::API
   def with_current_request
     Insights::API::Common::Request.with_request(request) do |current|
       Rails.logger.info("Request started #{request.original_url}")
-      if current.required_auth?
-        raise Insights::API::Common::EntitlementError, "User not Entitled" unless check_entitled(current.entitlement)
-
-        Thread.current[:user] = UserContext.new(Insights::API::Common::Request.current!, params)
-        ActsAsTenant.with_tenant(current_tenant(current)) { yield.tap { Rails.logger.info("Request ended #{request.original_url}") } }
-      else
-        ActsAsTenant.without_tenant { yield.tap {Rails.logger.info("Request ended #{request.original_url}")} }
+      UserContext.with_user_context(UserContext.new(current, params)) do |user_context|
+        if current.required_auth?
+          raise Insights::API::Common::EntitlementError, "User not Entitled" unless check_entitled(current.entitlement)
+
+          Thread.current[:user] = UserContext.new(Insights::API::Common::Request.current!, params)
```

Comment:
We don't need to do this anymore, right? Since it's handled inside the `.with_user_context` method?

Reply:
@eclarizio Fixed.